src/host/vsts.ts

```diff
@@ -8,7 +8,7 @@ export default class Vsts implements Host {
      * @example https://my-tenant.visualstudio.com/DefaultCollection/MyCollection/_git/my-repo
      * @example ssh://my-tenant@my-tenant.visualstudio.com:22/DefaultCollection/MyCollection/_git/my-repo
      */
-    private static urlRegex: RegExp = /(?:https:\/\/|ssh:\/\/.+@)(\w+)\.visualstudio\.com(?:\:\d+)?\/([/\w]+)\/_git\/([^/]+)/i;
+    private static urlRegex: RegExp = /(?:https:\/\/|ssh:\/\/.+@)(\w+)\.visualstudio\.com(?:\:\d+)?\/(.+)\/_git\/([^/]+)/i;
```

Comment:
`urlRegex` would be failed for the example you have provided, `ssh://my-tenant@my-tenant.visualstudio.com:22/DefaultCollection/MyCollection/_git/my-repo`